src/lsp_data.rs

```diff
@@ -256,23 +257,32 @@ impl RangeExt for Range {
     }
 }
 
+/// `DidChangeConfigurationParams.settings` payload reading the { rust: {...} } bit.
+#[derive(Debug, Deserialize)]
+pub struct ChangeConfigSettings {
+    pub rust: config::Config,
```

Comment:
I think this is an unfortunate leftover on how the rls-vscode by default automatically watched the `"rust" : {}"` key in user settings and passed it to the LSP client as a whole.

I'd love to get rid of the outer `"rust"` key but that's something that also needs to be done properly and coordinated in the rls-vscode, so this definitely shouldn't be blocked on this.

Reply:
It'd have to be coordinated in all client's, or backward compatible. So I don't think it's worth changing to be honest.